PirateParser/Parsers/ForLoopStatementParser.cs

```diff
@@ -0,0 +1,78 @@
+using PirateParser.Node;
+using PirateParser.Node.Interfaces;
+
+namespace PirateParser.Parsers;
+
+public class ForLoopStatementParser : BaseParser
+{
+    private ParserFactory _parserFactory { get; set; }
+
+    public ForLoopStatementParser(List<Token> tokens, Token token, ILogger logger, ParserFactory parserFactory) : base(tokens, token, logger)
```

Comment:
Think if its possible to pass in the list and define the current one at the same time.